tests/phpunit/tests/functions/wpNonceUrl.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+/**
+ * @group functions.php
+ *
+ * @covers ::wp_nonce_url
+ */
+class Tests_Functions_WpNonceUrl extends WP_UnitTestCase {
+	/**
+	 * Tests that wp_nonce_url() appends the nonce name and value to the URL.
+	 *
+	 * @dataProvider data_should_append_nonce_name_and_value
+	 *
+	 * @ticket 54870
+	 *
+	 * @param string     $actionurl URL to add nonce action.
+	 * @param int|string $action    Optional. Nonce action name. Default -1.
+	 * @param string     $name      Optional. Nonce name. Default '_wpnonce'.
+	 */
+	public function test_should_append_nonce_name_and_value( $actionurl, $action = -1, $name = '_wpnonce' ) {
+		$actual        = wp_nonce_url( $actionurl, $action, $name );
+		$url_with_name = "$actionurl?$name=";
+		$nonce         = str_replace( $url_with_name, '', $actual );
+
+		$this->assertStringContainsString(
+			$url_with_name,
+			$actual,
+			'The URL did not contain the action URL and the nonce name'
+		);
+
+		$this->assertNotFalse(
+			wp_verify_nonce( $nonce, $action ),
+			'The nonce is invalid'
+		);
+	}
+
+	/**
+	 * Data provider
```

Comment:
```suggestion
	 * Data provider for test_should_append_nonce_name_and_value().
```

Reply:
Fixed in [5be5a43](https://github.com/WordPress/wordpress-develop/pull/2653/commits/5be5a4392bfffc703362ae3fde4abfbff63f04d2).